doc/developer-notes.md

```diff
@@ -513,6 +515,49 @@ To enable test parallelism:
 cmake -DJOBS=$(nproc) -P build/Coverage.cmake
 ```
 
+#### Using LLVM/Clang toolchain
+
+The following generates a coverage report for unit tests.
+
+Configure the build with the following flags:
+
+```shell
+# MacOS may require -DCMAKE_CXX_COMPILER="$(brew --prefix llvm)/bin/clang++" and -DCMAKE_C_COMPILER="$(brew --prefix llvm)/bin/clang"
+cmake -B build -DCMAKE_C_COMPILER="clang" \
+   -DCMAKE_CXX_COMPILER="clang++" \
+   -DCMAKE_C_FLAGS="-fprofile-instr-generate -fcoverage-mapping" \
+   -DCMAKE_CXX_FLAGS="-fprofile-instr-generate -fcoverage-mapping"
+cmake --build build # Use "-j N" here for N parallel jobs.
+```
+
+Generating the raw profile data based on `ctest` execution:
+
+```shell
+LLVM_PROFILE_FILE="default_%p.profraw" ctest --test-dir build # Use "-j N" here for N parallel jobs.
+
+# merge all the raw profile data into a single file
+find build -name "default_*.profraw" | xargs llvm-profdata merge -sparse -o build/coverage.profdata
+# Note: The "counter mismatch" warning can be safely ignored, though it can be resolved by updating to Clang 19.
+# The warning occurs due to version mismatches but doesn't affect the coverage report generation.
+```
+
+Generating the coverage report:
+
+```shell
+llvm-cov show build/src/test/test_bitcoin \
+    --instr-profile=build/coverage.profdata \
+    --format=html \
+    --show-instantiation-summary \
+    --show-line-counts-or-regions \
+    --show-expansions \
+    --output-dir=build/coverage_report \
+    --project-title="Bitcoin Core Coverage Report"
```

Comment:
If there are profile files from more than one executable, like from `test_bitcoin` and from `bitcoind` then the syntax of this changes a little bit: drop `build/src/test/test_bitcoin` and add `-object=build/src/test/test_bitcoin -object=build/src/bitcoind`:

```
llvm-cov show 
    --object=build/src/test/test_bitcoin \
    --object=build/src/bitcoind \
    --instr-profile=build/coverage.profdata \
    --format=html \
    --show-instantiation-summary \
    --show-line-counts-or-regions \
    --show-expansions \
    --output-dir=build/coverage_report \
    --project-title="Bitcoin Core Coverage Report"
```